### DISCUSSION TITLE
Dodać EAN to sale/offers

### DISCUSSION THREAD

**Discussion by haloween** [allegro/allegro-api#4327] [Suggestions - rejected]
:bulb: **Sugestia**
To nie powinno spowodować jakiegoś drastycznego pogorszenia wydajności a może zmniejszyć liczbę potrzebnych zapytań do API.
Scenariusz:
- oprogramowanie zewnętrzne wystawia Aukcje z EAN i ExtelnalId
- żeby pobrać EAN muszę każdorazowo klepać details tej aukcji, EANY się zmieniają (są producenci którzy tak robią)

**Comment by MaciejFrackowiak** [comment]
Pole "ean" niebawem zostanie usunięt z modelu oferty - szczegóły znajdziesz w [poście](https://github.com/allegro/allegro-api/issues/4042).

**Comment by jmarek-pl** [comment]
Chciałbym pobrać do bazy danych EAN-y (prawidłowa nazwa GTIN) z ofert wystawianych przez klienta, czyli przez sale/offers. Aktualnie udało mi się pobrać listę wszystkich oferta, a następnie odpytywać poszczególne id ofert w celu zwróceniu pełnych danych oferty czyli przez sale/offers/{id}. Tam znalazłem pole ean. Działa to dosyć wolno. W przyszłości chciałbym poprawiać lub nadać przez API te EAN-y na ofertach. Jak powinienem to robić biorąc pod uwagę zmiany w API Allegro i czy da się szybciej.

**Comment by MaciejFrackowiak** [comment]
@jmarek-pl Pole "ean" zostanie usunięte 29 marca 2021 r. z modelu oferty. Wartość EAN będzie dostępna tylko w parametrze GTIN - jego szczegóły pobierzesz wyłącznie zasobem [GET /sale/offers/{offerId}](https://developer.allegro.pl/documentation/#operation/getOfferUsingGET) - nie ma innej drogi "na skróty".

**Comment by jmarek-pl** [comment]
Dopiero teraz zauważyłem to pole "parameters", czyli mam tablicę do przeszukania w celu znalezienia EAN/GTIN. Jeśli dobrze rozumiem to trzeba szukać rekordu o id=225693. Następnie w polu values widzę kolejną tablicę z tym EAN-em.

**Comment by MaciejFrackowiak** [comment]
@jmarek-pl Parametr GTIN występuje w jednej z trzech postaci, w zależności od kategorii:
- EAN (id parametru=225693),
- ISBN (id parametru=245669),
- ISSN (id parametru=245673).

I to w tym parametrze znajdziesz obecnie i w przyszłości wartość EAN/ISBN/ISSN.
